### ISSUE TITLE
Brushless checkbox only changes plan options, not EBB connection

### ISSUE THREAD

**Issue by alexrudd2** [alexrudd2/saxi#126]
https://github.com/nornagon/saxi/pull/195 had a case where starting the app via `npm start`, and then later clicking the 'brushless' checkbox in the UI didn't apple the changes to switch to brushless. Writing a test to catch this case would be useful, and fixing whatever is broken.

originally posted by @jedahan in https://github.com/nornagon/saxi/issues/202#issue-1991183649

**Comment by alexrudd2** [comment]
I can reproduce.  The PlanOptions and so on get changed by toggling the checkbox. However, there's currently no way of changing the underlying EBB driver. It gets initialized once and then can't be changed.

There is mismatch in data flow.
The UI checkbox is connected to `planOptions.hardware` in `ui.tsx` and `planning.ts`
The CLI argument is connected to `ebb.hardware` in `ebb.ts` and  `server.ts`.

The EBB device info used to be sent all the way up to the UI (in https://github.com/nornagon/saxi/commit/0e9cd16d7c2276bb1c11bca045f19594e1e834f5).  However, that broke with the refactor in https://github.com/alexrudd2/saxi/commit/3e89b8556ec1256559fabb68a0b1fbca8cbc1049, since the new `getDeviceInfo()` change the signature.

**Comment by alexrudd2** [comment]
@jedahan does my explanation make sense?  any ideas how to clean it up?

**Comment by jedahan** [comment]
I had a draft comment from a while ago, but I will need to spend some time re-familiarizing myself with the code to see if this is off-tangent or relevant. Expect a followup in ~3 days

---

Some ideas for refactors. There are a few things going on here - one is properly bringing hardware outside planOptions in the UI (this might have just been a hack for me to re-use existing codepaths, instead of understanding the UI framework better), and a way to move initialization of things like pin settings to runtime instead of "build" time.

The second thing is that I think the word hardware is overloaded. We could jam on a document to explore options, but here is just one way to slice things, to get that discussion started, as an example:

* board - should be consistent across plotters that conform to the ebb command set, and include the ability to communicate between the host and board
* toolhead - brushless vs non-brushless, any other pinouts we care about **(do speed setting live here as well?)**
* plotter - board + toolhead (this could be presets, or its own model, and/or be called 'hardware') **(do speed settings live here?) **

* settings - settings for a particular plot
* plot - source image + settings

* plan - moves sent to plotter by applying settings to a vector image

---

**Comment by alexrudd2** [comment]
I took another look and created a (very rough) PR that does manage to change the underlying `ebb.hardware` (i.e. it changes the pin used).  Not sure how best to combine with the planner and React checkbox,

**Comment by alexrudd2** [comment]
> The second thing is that I think the word hardware is overloaded. We could jam on a document to explore options, but here is just one way to slice things, to get that discussion started, as an example:

Good point.  What about keeping things simple and just using `motor`?

**Comment by alexrudd2** [comment]
> I took another look and created a (very rough) PR that does manage to change the underlying `ebb.hardware` (i.e. it changes the pin used). Not sure how best to combine with the planner and React checkbox,

Yeah, I realize there is currently no mechanism that updates the React checkbox, since it's storing its state rather than getting it from the EBB.

`state.deviceInfo.hardware` will have information on the EBB.... eventually.  The checkbox is drawn before the websocket message `dev` is sent and `ondevinfo()` fires.

**Comment by philipp-lehmann** [comment]
Is there any progress on that topic. Is it possible to manually change to the brushless option when running `npm run dev`?

Thanks for providing the tool. It has been a game changer for me. Would be awesome if I could still use it after the upgrade.

**Comment by alexrudd2** [comment]
Sorry, there has not yet been progress on the _button_.  However, you can still use the brushless motor with the command-line argument.  `--hardware brushless`

https://github.com/alexrudd2/saxi/blob/d82524154094698d109f513d47c107c6035692b5/src/cli.ts#L23-L27

**Comment by philipp-lehmann** [comment]
Thanks a lot.

**Comment by philipp-lehmann** [comment]
<img width="600" alt="Image" src="https://github.com/user-attachments/assets/e2f84970-7236-4102-8f3e-85fe28844154" />

Unfortunately saxi is still falling back to the v3 version. 
(I don't expect a fix, just letting people know when looking for answers)

**Comment by alexrudd2** [comment]
Does the plotting work?

I can never remember if it's "--hardware brushless" or "hardware=brushless", but am able to plot with a brushless motor.

**Comment by alexrudd2** [comment]
Getting closer to understanding why the display shows `v3` despite being configured for brushless

![Image](https://github.com/user-attachments/assets/65ea7840-4151-4294-ba22-f73d969b16fc)

`planOptions.deviceInfo.hardware` !== `planOptions.hardware`

**Comment by alexrudd2** [comment]
@philipp-lehmann 
As of https://github.com/alexrudd2/saxi/commit/80cfa1bc1fbecd369ce272b99f05c10494c717ac, the text state will reflect the configured command line option.  `npm run start -- --hardware=brushless` will show `motor: brushless` in the UI and send the correct commands to the EBB.
(Since this is in `main` you can get it with `git pull`)

You could also use https://github.com/alexrudd2/saxi/pull/133/commits/f785eebc844611062553cf57c9023f6762ad5f5a, which brings back the checkbox as a way to change the motor after (or instead of) the CLI setting.  If the box is (un)checked, the relevant commands will be sent.
(`git fetch && git checkout f785eebc844611062553cf57c9023f6762ad5f5a`

**Comment by philipp-lehmann** [comment]
@alexrudd2 Btw it works well with [f785eeb](https://github.com/alexrudd2/saxi/commit/f785eebc844611062553cf57c9023f6762ad5f5a) 
Many thanks.
